packages/babel-generator/src/generators/flow.js

```diff
@@ -221,11 +221,7 @@ export function TypeAnnotation(node: Object) {
 }
 
 export function TypeParameter(node: Object) {
-  if (node.variance === "plus") {
-    this.token("+");
-  } else if (node.variance === "minus") {
-    this.token("-");
-  }
```

Comment:
For the generator specifically we'd probably be able to do

```
if (node.variance && typeof node.variance === 'object'){
  this.print(node.variance, node);
} else if (node.variance === "plus"){ // ...
```

my concern is more related to my other comment about the `visitor` list, because if we add `variance` to the `visitor` list for `TypeParameter`, people using older versions of Babylon and getting older ASTs will start getting errors because traversal will start traversing down into the string value expecting it to be an AST node.